File: cash.c
#include <stdio.h>
#include <cs50.h>
#include <math.h>

int main(void)
{
    float cash;

    do
    {
        cash = get_float("Enter amount of change to be counted: ");
    }
    while (cash <= 0);
    // round float and convert to int
    int cents;
    cents = roundf(cash * 100);


    // count the number of coins
    int counter;
    counter = 0;

    // find number of quarters
    while (cents >= 25)
    {
        counter += 1;
        cents = cents - 25;
    }

    // find number of dimes
    while (cents >= 10)
    {
        cents = cents - 10;
        counter += 1;
    }

    // find number of nickels
    while (cents >= 5)
    {
        cents = cents - 5;
        counter += 1;
    }

    // find number of pennies
    while (cents < 5 && cents > 0)
    {
        cents = cents - 1;
        counter += 1;
    }

    printf("%d\n", counter);

    return 0;
}

File: pset6/cash/cash.py
import cs50

# get and validate input
while True:
    dollars = cs50.get_float("Enter amount of change to be counted: ")
    if dollars > 0:
        break

#convert dollars to cents and round
cents = round(dollars * 100)
coins = 0

while cents > 0:
    #count quarters
    if cents >= 25:
        cents -= 25
        coins += 1

    #count dimes
    elif cents >= 10:
        cents -= 10
        coins += 1

    #count nickels
    elif cents >= 5:
        cents -= 5
        coins += 1

    #add remaining pennies
    elif cents >= 1:
        cents -= 1
        coins += 1

#print result
print(coins)